app/code/Magento/Payment/Model/Method/Adapter.php

```diff
@@ -508,11 +508,12 @@ private function executeCommand($commandCode, array $arguments = [])
         }
 
         if (isset($arguments['payment'])) {
+            $payment = $arguments['payment'];
             $arguments['payment'] = $this->paymentDataObjectFactory->create($arguments['payment']);
-        }
-
-        if ($this->commandExecutor !== null) {
-            return $this->commandExecutor->executeByCode($commandCode, $arguments);
+            
+            if ($this->commandExecutor !== null) {
```

Comment:
Hey,
  Hmm.. I've actually considered that - the problem is, with the current interface it's not possible to call executeByCode without passing it a valid payment object. So in case payment is not provided in the arguments, we shouldn't call executeByCode at all - or the CommandManagerInterface interface should be changed.